src/LsmTree.hs

```diff
@@ -33,14 +34,17 @@ insertTree x tree = case tree of
         False -> TreeNode l val (insertTree x r)
 
 -- | Data type that stores values in memory.
-data Table a = Table [a]
-    deriving stock (Eq, Show)
+data Table = Table
+    { fileName :: String
```

Comment:
There's `FilePath` type which is just an alias to `String` but usually it's better to use it. And in future, if you want, you can use a fancier type.